modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/snapshot/SnapshotExecutorImpl.java

```diff
@@ -250,7 +250,13 @@ public boolean init(final SnapshotExecutorOptions opts) {
         final FirstSnapshotLoadDone done = new FirstSnapshotLoadDone(reader);
         Requires.requireTrue(this.fsmCaller.onSnapshotLoad(done));
         try {
-            done.waitForRun();
+            // TODO: IGNITE-19363 We want to avoid deadlock for now, but this is an ad-hoc decision.
+            // We don't wait for the partition's snapshot load closure to finish here.
+            if (!node.getNodeId().getGroupId().contains("part")) {
+                done.waitForRun();
+            } else {
+                done.status = new Status();
```

Comment:
How does this work? Don't we have a race here with setting the task status? What will happen if snapshot loading fails?

Reply:
Also, there's a `Status.OK()` method, which looks a little bit more descriptive

Reply:
This is a temporal solution, so we don't really care about failing loading the snapshot. The race doesn't matter, we won't be checking the result

Reply:
I really hope that it is temporary, not checking the status seems dangerous

Reply:
It is, I am going to start on fixing it right away